docs/environment.md
## Environment variables

Unlike traditional Django settings, we use environment variables
for configuration to be compliant with [twelve-factor][] apps.

You can define environment variables using your environment, or
(if you're developing locally) an `.env` file in the root directory
of the repository. For more information on configuring
environment variables on cloud.gov, see
[Deploying to Cloud Foundry](deploy.md).

**Note:** When an environment variable is described as representing a
boolean value, if the variable exists with *any* value (even the empty
string), the boolean is true; otherwise, it's false.

### Local development settings

These settings are likely to only be useful during local development
and/or the deployment of non-production instances of CALC.

* `DEBUG` is a boolean value that indicates whether debugging is enabled
  (this should always be false in production).

* `DEBUG_LOG_SQL` is a boolean value that indicates whether SQL
  statements sent to the database should be logged to the console.

* `DEBUG_HTTPS` is a boolean value that indicates whether the
  site should consider itself to be served over HTTPS while
  debugging is enabled. This can be useful if you want to develop
  with SSL enabled.

* `HIDE_DEBUG_UI` is a boolean value that indicates whether to hide
  various development and debugging affordances in the UI, such as the
  [Django Debug Toolbar][]. This can be useful when demoing or user testing
  a debug build.

* `REDIS_TEST_URL` is the redis URL to use when running tests.
  When `DEBUG` is true *and* `REDIS_URL` isn't defined, it defaults to
  `redis://localhost:6379/1`.

* `FORCE_DISABLE_SECURE_SSL_REDIRECT` is a boolean value that indicates
  whether to disable redirection from http to https. Because such
  redirection is enabled by default when `DEBUG` is false, this option
  can be useful when you want to simulate *almost* everything about a
  production environment without having to setup SSL.

* `USE_POLLING` is a boolean; if true, this will force all the watchers
  to use filesystem polling instead of OS notifications, which works
  better with some configurations, such as Windows, VirtualBox, and
  NFS mounts.

* `NON_PROD_INSTANCE_NAME` is an optional instance name that when specified
  will cause a banner to be shown at the top of every page to let users know
  that they are viewing a non-production instance of CALC. This value
  can contain HTML, so it's possible to e.g. wrap the value in a hyperlink.

### Deployment settings

These are useful for any kind of deployment of CALC.

* `SECRET_KEY` is a large random value corresponding to Django's
  [`SECRET_KEY`][] setting. It is automatically set to a known, insecure
  value when `DEBUG` is true.

* `DATABASE_URL` is the URL for the database, as per the
  [DJ-Database-URL schema][]. Note that the protocol *must* be
  `postgres:`.

* `EMAIL_URL` is the URL for the service to use when sending
  email, as per the [dj-email-url schema][]. When `DEBUG` is true,
  this defaults to `console:`. If it is set to `dummy:` then no emails will
  be sent and messages about email notifications will not be shown to users.
  The setting can easily be manually tested via the `manage.py sendtestemail`
  command.

* `DEFAULT_FROM_EMAIL` is the email from-address to use in all system
  generated emails to users. It corresponds to Django's [`DEFAULT_FROM_EMAIL`][]
  setting. It defaults to `noreply@localhost` when `DEBUG=True`.

* `SERVER_EMAIL` is the email from-address to use in all system generated
  emails to managers and admins. It corresponds to Django's [`SERVER_EMAIL`][]
  setting. It defaults to `system@localhost` when `DEBUG=True`.

* `HELP_EMAIL` is the email  used as the reply-to address in system
  generated emails to users. It is also the email address used in the site
  footer and for other contact purposes. It should refer to an inbox that is
  monitored. If not set, it will use the same value as `DEFAULT_FROM_EMAIL`.

* `REDIS_URL` is the URL for redis, which is used by the task queue.
  When `DEBUG` is true, it defaults to `redis://localhost:6379/0`.

* `ENABLE_SEO_INDEXING` is a boolean value that indicates whether to
  indicate to search engines that they can index the site.

* `UAA_CLIENT_ID` is your cloud.gov/Cloud Foundry UAA client ID. It
  defaults to `calc-dev`.

* `UAA_CLIENT_SECRET` is your cloud.gov/Cloud Foundry UAA client secret.
  If this is undefined and `DEBUG` is true, then a built-in Fake UAA Provider
  will be used to "simulate" cloud.gov login.

* `SECURITY_HEADERS_ON_ERROR_ONLY` is a boolean value that indicates whether
  security-related response headers (such as `X-XSS-Protection`)
  should only be added on error (status code >= 400) responses. This setting
  will likely only be used for cloud.gov deployments, where the built-in proxy
  sets those security headers on 200 responses but not on others.

* `GA_TRACKING_ID` is the tracking ID (e.g. `'UA-12345678-12'`)
  for the associated Google Analytics account.
  It will default to the empty string if not found in the environment.

* `SLACKBOT_WEBHOOK_URL` is the URL of a [Slack incoming webhook][] that
  will be sent messages whenever certain kinds of
  [events](../slackbot/signals.py) occur in the app.

[`SECRET_KEY`]: https://docs.djangoproject.com/en/1.8/ref/settings/#secret-key
[`DEFAULT_FROM_EMAIL`]: https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-DEFAULT_FROM_EMAIL
[`SERVER_EMAIL`]: https://docs.djangoproject.com/en/1.8/ref/settings/#std:setting-SERVER_EMAIL
[Django Debug Toolbar]: https://github.com/jazzband/django-debug-toolbar/
[DJ-Database-URL schema]: https://github.com/kennethreitz/dj-database-url#url-schema
[dj-email-url schema]: https://github.com/migonzalvar/dj-email-url#supported-backends
[twelve-factor]: http://12factor.net/
[Slack incoming webhook]: https://api.slack.com/incoming-webhooks


CHANGELOG.md
# Change log

All notable changes to this project will be documented in this file,
which uses the format described in
[keepachangelog.com](http://keepachangelog.com/). This project adheres
to [Semantic Versioning](http://semver.org/).

## [Unreleased][unreleased]

## 2020-05-28

### Changed

- Repalced Price Prosal Template (PPT) for Mas consolidated contract vehicle


## 2020-05-04

### Changed

- Added additional(optional) search filters based on Sin mumber, Security Clearance 
- Upgraded snyk issues
- Addressed Bugs


## 2020-04-11

### Changed

- Added new search filter based on Keywords, Description, Certifications
- Added COVID-19 banner 
- Upgraded high and medium severity vulnerabilities (reported by snyk tool)


## 2020-03-18

### Changed

- More data clean up on data quality report
- Upgraded high and medium severity vulnerabilities (reported by snyk tool)


## 2020-02-25

### Changed

- Expired contracts and duplicated records were removed for the data quality report
- Upgraded 1 high severity on django (reported by snyk tool)


## 2020-01-27

### Changed

- Upgraded packages that is causing severity vulnerabilities.


## 2019-12-06

### Changed

- Fixes of Snyk 9 high severity and 5 medium severity vulnerabilities.

## 2019-11-19

### Changed

- Fix JavaScript issue of price list upload on Chrome browser.
- Add disclaimer on CALC under the Exact match box.
- Calc About page updates.

## 2019-11-19

### Changed

- Admin users has ability to upload Capability Statement file in CALC.
- Admin users has ability to add URL of a capability statement in CALC.
- Created API to get Capability statements. Any application can use this service to retriever capability statements by contact number.
- GSA API Standards Adoption for CALC.

## 2019-08-22

### Changed

- Performed FISMA Annual Self-Assessment CALC.


## 2019-10-08

### Changed

- Added Option to Upload the MAS Consolidation Pricing Template. (#2360)
- 1 high severity issues fixed (Reported by Synk tool).


## 2019-08-20

### Changed

- Update content on “about CALC” page. (#2317)
- What does CALC do?
- How can CALC help me?
- Where do the prices in this tool come from?
- Explanation of +/-1 standard deviation under CALC’s visualization

## 2019-07-12

### Added
- Price list upload feature for Schedule 71IIK. (#2310)

## 2019-07-03

### Changed
- 1 high severity issue fixed (Reported by Synk tool. Upgraded Django version to 1.11.21). (#2303)

## 2019-06-25

### Changed
- 2 high severity issues fixed (Reported by Synk tool). (#2300)

## 2019-06-20

### Added
- Implement Session Timeout For Admin Users. (#2298)

## 2019-05-23

### Changed
- 1 high severity issue fixed (Reported by Synk tool). (#2284)

## 2019-05-16

### Changed
- Update Linux instances to cflinuxfs3 from cflinuxfs2. (#2281)

## 2019-04-30

### Changed
- 1 high severity issue fixed (Reported by Synk tool). (#2277)

## 2019-04-19

### Changed
- 3 high severity and 2 medium severity issues fixed (Reported by Synk tool). (#2275)

## 2019-04-04

### Changed
- Removed New Relic from Calc App. (#2267)

## 2019-03-29

### Changed
- Removed Beta Tag from master. (#2264)

## 2019-03-07

### Changed
- 1 high severity issue fixed (Reported by Synk tool). (#2262)

## 2019-02-22

### Changed
- RQ Scheduler Version Upgrade. (#2260)

## 2019-02-08

### Changed
- 6 security issues fixed (Reported by Synk tool). (#2262)

## [3.2.2][] - 2018-09-28

### Changed
- Add NLTK's Averaged Perceptron Tagger `.pickle` file to repository instead of
  downloading it on every build, and make changes to build files and to price
  analysis reflecting those changes. (#2194)

## [3.2.1][] - 2018-09-27

### Added
- Add deployment step of NLTK resources to support price list analysis feature. (#2192)

### Changed
- Fixed a bug in lists of labor categories that are too long for proper visual display. (#2191)

## [3.2.0][] - 2018-09-27

### Added
- Schedule 70 contracting officers with login credentials can now have CALC
  analyze price lists and provide summary stats about them. (#997)
- The data upload process now has an overview and walkthrough of the process
  on the first page. (#1460)

### Changed
- Fixed a bug that caused CircleCI to time out on Jest tests. (#2149, #2155)
- Moved the data explorer's proposed price entry box down to be with the
  rest of the pricing summary numbers. (#2178)
- Changed the data explorer's summary hierarchy (#2178)
- Updated the design of the data explorer graph to match site colors. (#2143)
- Made some updates and language changes to the deployment documentation. (#2139)
- Removed some unused template tags from the header in the admin section that
  were rendering oddly. (#2162)
- Fixed a bug that made the "exact match" checkbox too wide. (#2141)

## [3.1.0][] - 2018-09-19

### Added
- Users can now search by vendor name or contract number in addition to labor category (#2069).
- The data upload tool now accepts Schedule 736 uploads (#2140).

## [3.0.1][] - 2018-09-04

### Changed
- Vendor name and contract number have been consolidated into a single column for easier scanning (#2112).
- Schedule selector search UI has been relocated and improved (#2085).
- Data administrators can now access the admin panel from the user menu rather than the main navigation (#2105).
- Search type UI has been simplified to a single checkbox (#2108).

### Added
- API documentation is now available in the CALC footer (#2065).

## [3.0.0][] - 2018-08-20

### Changed
- Changes to the options available for search. `match_phrase` has been removed as an
option due to observed limited utility and bugginess in the implementation. Any API
calls to `query_type` relying on `match_phrase` will return `match_words` results instead. The
`match_words` and `match_exact` options have also been refactored to remove outstanding bugs. (#2061)
- Updated UI to reflect search API changes (#2090).
- Fixed an issue with front end files not being built properly (#2088).

calc/settings.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-
# mypy: ignore-errors
"""
Django settings for calc project.
For more information on this file, see
https://docs.djangoproject.com/en/1.7/topics/settings/
For the full list of settings and their values, see
https://docs.djangoproject.com/en/1.7/ref/settings/
"""

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
import os
os.environ.setdefault("DJANGO_SETTINGS_MODULE", __file__)
import dj_database_url
import dj_email_url
from dotenv import load_dotenv
from typing import Tuple, Any, Dict  # NOQA
from .settings_utils import (load_cups_from_vcap_services,
                             load_redis_url_from_vcap_services,
                             is_running_tests)

BASE_DIR = os.path.dirname(os.path.dirname(__file__))

DOTENV_PATH = os.path.join(BASE_DIR, '.env')

if os.path.exists(DOTENV_PATH):
    load_dotenv(DOTENV_PATH)

load_cups_from_vcap_services()
load_redis_url_from_vcap_services('calc-redis32')
NON_PROD_INSTANCE_NAME = os.environ.get('NON_PROD_INSTANCE_NAME', '')

print('heres the non prod instance name', NON_PROD_INSTANCE_NAME)
# SECURITY WARNING: don't run with debug turned on in production!
# if NON_PROD_INSTANCE_NAME == 'staging':
if NON_PROD_INSTANCE_NAME == 'staging':
    DEBUG = True
else:
    DEBUG = 'DEBUG' in os.environ

DEBUG_HTTPS = 'DEBUG_HTTPS' in os.environ and not is_running_tests()

HIDE_DEBUG_UI = 'HIDE_DEBUG_UI' in os.environ

SLACKBOT_WEBHOOK_URL = os.environ.get('SLACKBOT_WEBHOOK_URL', '')

if is_running_tests():
    HIDE_DEBUG_UI = True
    SLACKBOT_WEBHOOK_URL = ''

if DEBUG:
    os.environ.setdefault(
        'SECRET_KEY',
        'I am an insecure secret key intended ONLY for dev/testing.'
    )
    os.environ.setdefault(
        'EMAIL_URL',
        os.environ.get('DEFAULT_DEBUG_EMAIL_URL', 'console:')
    )
    if 'REDIS_URL' not in os.environ:
        # Only set a default REDIS_TEST_URL if REDIS_URL is not
        # explicitly defined either.
        os.environ.setdefault('REDIS_TEST_URL', 'redis://localhost:6379/1')
    os.environ.setdefault('REDIS_URL', 'redis://localhost:6379/0')
    os.environ.setdefault('DEFAULT_FROM_EMAIL', 'noreply@localhost')
    os.environ.setdefault('SERVER_EMAIL', 'system@localhost')

if 'EMAIL_URL' not in os.environ:
    raise Exception('Please define the EMAIL_URL environment variable!')

SEND_TRANSACTIONAL_EMAILS = os.environ['EMAIL_URL'] == 'dummy:'

email_config = dj_email_url.config()
# Sets a number of settings values, as described at
# https://github.com/migonzalvar/dj-email-url
vars().update(email_config)

DEFAULT_FROM_EMAIL = os.environ['DEFAULT_FROM_EMAIL']
SERVER_EMAIL = os.environ['SERVER_EMAIL']
HELP_EMAIL = os.environ.get('HELP_EMAIL', DEFAULT_FROM_EMAIL)

GA_TRACKING_ID = os.environ.get('GA_TRACKING_ID', '')

TEMPLATES = [{
    'BACKEND': 'django.template.backends.django.DjangoTemplates',
    'DIRS': [
        os.path.join(BASE_DIR, 'calc/templates'),
    ],
    'APP_DIRS': True,
    'OPTIONS': {
        'context_processors': [
            'calc.context_processors.show_debug_ui',
            'calc.context_processors.google_analytics_tracking_id',
            'calc.context_processors.help_email',
            'calc.context_processors.non_prod_instance_name',
            'calc.context_processors.sample_users',
            'calc.context_processors.api_host',
            'frontend.context_processors.is_safe_mode_enabled',
            "django.contrib.auth.context_processors.auth",
            "django.template.context_processors.debug",
            "django.template.context_processors.i18n",
            "django.template.context_processors.media",
            'django.template.context_processors.request',
            "django.template.context_processors.static",
            "django.template.context_processors.tz",
            "django.contrib.messages.context_processors.messages",
            "django.template.context_processors.request",
        ],
    },
}]

ALLOWED_HOSTS = ['*']

BASE_GITHUB_URL = 'https://github.com/18F/calc'

DATA_CAPTURE_APP_CONFIG = 'DefaultDataCaptureApp'

# When IS_RQ_SCHEDULER is in the env,
# instead use the special DataCaptureSchedulerApp since this process
# is being used as the scheduler instance.
if 'IS_RQ_SCHEDULER' in os.environ:
    DATA_CAPTURE_APP_CONFIG = 'DataCaptureSchedulerApp'


# Application definition

INSTALLED_APPS = (
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.humanize',
    'django.contrib.sites',
    'django.contrib.postgres',
    'whitenoise.runserver_nostatic',
    'django.contrib.staticfiles',
    'debug_toolbar',
    'django_rq',
    'session_security',
    'data_explorer',
    'contracts.apps.DefaultContractsApp',
    'data_capture.apps.{}'.format(DATA_CAPTURE_APP_CONFIG),
    'api',
    'rest_framework',
    'corsheaders',
    'uaa_client',
    'user_account',
    'styleguide',
    'meta',
    'frontend',
    'slackbot.apps.SlackbotConfig',
    'uswds_forms',
    'admin_reorder',
    'storages',
    'drf_yasg',
    # 'rest_framework_swagger'
)  # type: Tuple[str, ...]

SITE_ID = 1

if DEBUG:
    STATICFILES_STORAGE = 'frontend.crotchety.CrotchetyStaticFilesStorage'
    WHITENOISE_MIDDLEWARE = 'frontend.crotchety.CrotchetyWhiteNoiseMiddleware'
else:
    STATICFILES_STORAGE = ('whitenoise.storage.'
                           'CompressedManifestStaticFilesStorage')
    WHITENOISE_MIDDLEWARE = 'whitenoise.middleware.WhiteNoiseMiddleware'

SWAGGER_SETTINGS = {
    'FETCH_SCHEMA_WITH_QUERY': False,
    'SUPPORTED_SUBMIT_METHODS': ['post'],
    'SECURITY_DEFINITIONS': {
        'basic': {
            'type': 'basic'
        }
    },
}

MIDDLEWARE_CLASSES = (
    'django.middleware.cache.UpdateCacheMiddleware',
    'calc.middleware.ComplianceMiddleware',
    WHITENOISE_MIDDLEWARE,
    'corsheaders.middleware.CorsMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',

    'ip_restriction.IpWhitelister',

    # Temporary! 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'uaa_client.middleware.UaaRefreshMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'session_security.middleware.SessionSecurityMiddleware',
    # DjDT needs to be at the end of the middleware stack or else it can
    # cause issues with other middlewares' process_view methods
    # when the ProfilingPanel is enabled
    # http://django-debug-toolbar.readthedocs.io/en/stable/panels.html#profiling
    'calc.middleware.DebugOnlyDebugToolbarMiddleware',
    'django.middleware.cache.FetchFromCacheMiddleware',
    'admin_reorder.middleware.ModelAdminReorder',
    'django.middleware.security.SecurityMiddleware'
)

AUTHENTICATION_BACKENDS = (
    'uaa_client.authentication.UaaBackend',
)

ROOT_URLCONF = 'calc.urls'

WSGI_APPLICATION = 'calc.wsgi.application'

# Internationalization
# https://docs.djangoproject.com/en/1.7/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True

# django cors headers
CORS_ORIGIN_ALLOW_ALL = True
CORS_URLS_REGEX = r'^/api/.*$'  # only allow CORS for /api/ routes
CORS_ALLOW_METHODS = ('GET', 'OPTIONS',)  # only allow read-only methods

API_HOST = os.environ.get('API_HOST', '/api/')

# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/1.7/howto/static-files/
STATIC_URL = '/static/'
STATIC_ROOT = os.path.join(BASE_DIR, 'static')

STATICFILES_DIRS = [
    # os.path.join(BASE_DIR, 'static'),
    os.path.join(BASE_DIR, 'docs', 'static')
]

RQ_QUEUES = {
    'default': {
        'URL': os.environ['REDIS_URL'],
    }
}

if is_running_tests():
    RQ_QUEUES['default']['URL'] = os.environ['REDIS_TEST_URL']

PAGINATION = 200

REST_FRAMEWORK = {
    'COERCE_DECIMAL_TO_STRING': False,
    'DEFAULT_SCHEMA_CLASS': 'rest_framework.schemas.coreapi.AutoSchema'
}

LOGGING: Dict[str, Any] = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] "
                      "%(message)s",
            'datefmt': "%d/%b/%Y %H:%M:%S"
        },
        'simple': {
            'format': '%(levelname)s %(message)s'
        },
    },
    'handlers': {
        'file': {
            'level': 'DEBUG',
            'class': 'logging.FileHandler',
            'filename': os.path.join(BASE_DIR, 'logs/calc.log'),
            'formatter': 'verbose'
        },
        'contracts_file': {
            'level': 'DEBUG',
            'class': 'logging.FileHandler',
            'filename': os.path.join(BASE_DIR, 'logs/load_data.log'),
            'formatter': 'verbose'
        },
        'console': {
            'level': 'INFO',
            'class': 'logging.StreamHandler',
            'formatter': 'verbose'
        },
    },
    'loggers': {
        'django': {
            'handlers': ['console', 'file'],
            'propagate': True,
            'level': 'INFO',
        },
        'uaa_client': {
            'handlers': ['console', 'file'],
            'propagate': True,
            'level': 'INFO',
        },
        'calc': {
            'handlers': ['console', 'file'],
            'propagate': True,
            'level': 'INFO',
        },
        'contracts': {
            'handlers': ['console', 'contracts_file'],
            'propagate': True,
            'level': 'INFO',
        },
        'rq.worker': {
            'handlers': ['console'],
            'level': 'INFO',
        },
        'rq_scheduler': {
            'handlers': ['console'],
            'level': 'INFO',
        },
        'slackbot': {
            'handlers': ['console'],
            'level': 'INFO',
        },
    },
}

DEBUG_LOG_SQL = 'DEBUG_LOG_SQL' in os.environ

if DEBUG_LOG_SQL:
    LOGGING['handlers']['debug_console'] = {
        'level': 'DEBUG',
        'class': 'logging.StreamHandler',
        'formatter': 'verbose'
    }
    LOGGING['loggers']['django.db.backends'] = {
        'handlers': ['debug_console'],
        'level': 'DEBUG',
    }

DATABASES = {}
DATABASES['default'] = dj_database_url.config()
POSTGRES_VERSION = '9.5.4'

SECURE_SSL_REDIRECT = not DEBUG

if DEBUG and DEBUG_HTTPS:
    SECURE_SSL_REDIRECT = True

if 'FORCE_DISABLE_SECURE_SSL_REDIRECT' in os.environ:
    SECURE_SSL_REDIRECT = False

SESSION_COOKIE_SECURE = CSRF_COOKIE_SECURE = SECURE_SSL_REDIRECT
SESSION_EXPIRE_AT_BROWSER_CLOSE = True

CSRF_COOKIE_HTTPONLY = True

# Amazon ELBs pass on X-Forwarded-Proto.
SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")

# Amazon also sets X-Forwarded-Host.
USE_X_FORWARDED_HOST = True

SECRET_KEY = os.environ['SECRET_KEY']

ENABLE_SEO_INDEXING = 'ENABLE_SEO_INDEXING' in os.environ

SECURITY_HEADERS_ON_ERROR_ONLY = 'SECURITY_HEADERS_ON_ERROR_ONLY' in os.environ

DATA_CAPTURE_SCHEDULES = (
    'data_capture.schedules.region_10.Region10PriceList',
    'data_capture.schedules.s70.Schedule70PriceList',
    'data_capture.schedules.s03fac.Schedule03FACPriceList',
    'data_capture.schedules.s736.Schedule736PriceList',
    'data_capture.schedules.region_3.Region3PriceList',
    'data_capture.schedules.mas_consolidated.MASConsolidatedPriceList'
)  # type: Tuple[str, ...]

if DEBUG and not HIDE_DEBUG_UI:
    DATA_CAPTURE_SCHEDULES += (
        'data_capture.schedules.fake_schedule.FakeSchedulePriceList',
    )

if NON_PROD_INSTANCE_NAME == 'staging':
    UAA_AUTH_URL = "fake:"
    UAA_TOKEN_URL = "fake:"
    UAA_APPROVED_DOMAINS = ['gsa.gov', 'example.com']
    UAA_CLIENT_ID = 'fakeclientid'
    UAA_CLIENT_SECRET = 'fakeclientsecret'
    IPS = os.environ.get('WHITELISTED_IPS') if os.environ.get('WHITELISTED_IPS') else 'none,fool'
    print('heres the ips variable', IPS)
    RESTRICT_IPS = True
    ALLOWED_IPS = IPS.split(',')  # type: ignore
else:
    UAA_AUTH_URL = 'https://login.fr.cloud.gov/oauth/authorize'
    UAA_TOKEN_URL = 'https://uaa.fr.cloud.gov/oauth/token'
    UAA_CLIENT_ID = os.environ.get('UAA_CLIENT_ID', 'default_value')
    UAA_CLIENT_SECRET = os.environ.get('UAA_CLIENT_SECRET', 'default_value')

UAA_LOGOUT_URL = '/logout'

LOGIN_URL = 'uaa_client:login'

LOGIN_REDIRECT_URL = '/'

# We *always* want to send a Cache-Control header downstream, especially
# in the event where we've got a reverse proxy with aggressive caching
# defaults like Amazon CloudFront in front of us.
#
# For now we're just going to tell any downstream caches to never cache
# any dynamic content we give them, to ensure that stale content never
# gets served to end-users.
CACHE_MIDDLEWARE_SECONDS = 0
# seconds
SESSION_SECURITY_WARN_AFTER = 1500
# seconds
SESSION_SECURITY_EXPIRE_AFTER = 1800
SESSION_SECURITY_PASSIVE_URL_NAMES = ['ignore']

SESSION_EXPIRE_AFTER_LAST_ACTIVITY = True

if not UAA_CLIENT_SECRET:
    if DEBUG:
        # We'll be using the Fake UAA Provider.
        UAA_CLIENT_SECRET = 'fake-uaa-provider-client-secret'
        if not is_running_tests():
            UAA_AUTH_URL = UAA_TOKEN_URL = 'fake:'
            UAA_LOGOUT_URL = '/logout'

DEBUG_TOOLBAR_PATCH_SETTINGS = False

DEBUG_TOOLBAR_DISABLE_FOR_URL_PREFIXES = [
    '/styleguide/fullpage-example/',
]

DEBUG_TOOLBAR_CONFIG = {
    'DISABLE_PANELS': set([
        'debug_toolbar.panels.redirects.RedirectsPanel',
        'debug_toolbar.panels.profiling.ProfilingPanel',
    ]),
    'SHOW_TOOLBAR_CALLBACK': 'calc.middleware.show_toolbar',
}

TEMPLATE_CONTEXT_PROCESSORS = [
    'django.core.context_processors.request',
    'django.contrib.auth.context_processors.auth',
]

DEBUG_TOOLBAR_PANELS = [
    'debug_toolbar.panels.versions.VersionsPanel',
    'debug_toolbar.panels.profiling.ProfilingPanel',
    'debug_toolbar.panels.timer.TimerPanel',
    'debug_toolbar.panels.settings.SettingsPanel',
    'debug_toolbar.panels.headers.HeadersPanel',
    'debug_toolbar.panels.request.RequestPanel',
    'debug_toolbar.panels.sql.SQLPanel',
    'debug_toolbar.panels.staticfiles.StaticFilesPanel',
    'debug_toolbar.panels.templates.TemplatesPanel',
    'debug_toolbar.panels.cache.CachePanel',
    'debug_toolbar.panels.signals.SignalsPanel',
    'debug_toolbar.panels.logging.LoggingPanel',
    'debug_toolbar.panels.redirects.RedirectsPanel',
    'data_capture.panels.ScheduledJobsPanel',
]

PRICE_LIST_ANALYSIS_FINDERS = [
    'data_capture.analysis.finders.GteEduAndExpFinder',
]

if DEBUG:
    INSTALLED_APPS += (
        'django.contrib.admindocs',
    )

ADMIN_REORDER = (
    # Use django-modeladmin reorder to rearrange/rename the apps and models
    # https://pypi.org/project/django-modeladmin-reorder/
    {'app': 'data_capture', 'label': 'User-submitted pricing data', 'models': (
        {
            'model': 'data_capture.SubmittedPriceListRow',
            'label': 'Mute or unmute submitted price list rows'
        },
        {
            'model': 'data_capture.UnreviewedPriceList',
            'label': 'Approve or reject unreviewed price lists'
        },
        {
            'model': 'data_capture.ApprovedPriceList',
            'label': 'Retire approved price lists'
        },
        {
            'model': 'data_capture.RetiredPriceList',
            'label': 'Edit and re-approve retired price lists'
        },
        {
            'model': 'data_capture.RejectedPriceList',
            'label': 'Approve rejected price lists'
        },
        {
            'model': 'data_capture.AttemptedPriceListSubmission',
            'label': 'Replay uncompleted price list submission attempts'
        },
    )},
    {'app': 'contracts', 'label': 'Contracting Metadata', 'models': (
        {'model': 'contracts.ScheduleMetadata', 'label': 'Available schedules'},
    )},
    {'app': 'auth', 'label': 'Authentication and authorization', 'models': (
        'auth.User',
        {'model': 'auth.Group', 'label': 'User groups'},
    )},
    {'app': 'sites', 'label': 'Available site settings', 'models': (
        {'model': 'sites.Site', 'label': 'Site URLs'},
    )},
)

# NOT NEEDED FOR DEV
# AWS_ACCESS_KEY_ID = 'AKIAR7FXZINYORDIPBVD'
# AWS_SECRET_ACCESS_KEY = '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'
# AWS_STORAGE_BUCKET_NAME = 'cg-db8307d5-ae86-42ac-a205-7356d692d7d0'
# AWS_REGION = 'us-gov-west-1'
# AWS_S3_FILE_OVERWRITE = True
# AWS_DEFAULT_ACL = None
# DEFAULT_FILE_STORAGE = 'storages.backends.s3boto3.s3Boto3Storage'

AWS_ACCESS_KEY_ID = os.environ.get('ACCESS_KEY_ID')
AWS_SECRET_ACCESS_KEY = os.environ.get('SECRET_ACCESS_KEY')
AWS_STORAGE_BUCKET_NAME = os.environ.get('BUCKET')
AWS_REGION = os.environ.get('REGION')
AWS_S3_FILE_OVERWRITE = True
AWS_DEFAULT_ACL = None
DEFAULT_FILE_STORAGE = 'storages.backends.s3boto3.s3Boto3Storage'
